### PR TITLE
fix: linting issues, trailingCommas: all

### LINKED ISSUE #1033: Windows: Running yarn tests reports trying to use "printf"
Running the https://github.com/graphql/graphiql/blob/master/CONTRIBUTING.md#getting-started steps on windows 10 (command line running as admin) I get an error showing at the end of the run about `printf` which doesn't exist on windows. Not sure whats throwing it as the `yarn tests` command chains a bunch of stuff together so I'll run them individual and see if I can narrow it down.

### PR DESCRIPTION
**Pull request by acao**
addresses a number of annoying issues:

- prettier `trailingCommas` all (thats the big commit)
- `lint-staged` at the root
- `eslint --fix` and then`pretty --write`
- fix root commands as well
- `yarn lint` checks prettier and eslint rules for clashes

resolves #1033 by deprecating that statement for an alternative thats already built into resources/pretty.js and is implicitly multi-os

### CODE REVIEW

File: packages/codemirror-graphql/resources/watch.js [hunk 1]
```diff
@@ -101,14 +101,14 @@ function checkFiles(filepaths) {
     .then(testSuccess =>
       lintFiles(filepaths).then(lintSuccess =>
         typecheckStatus().then(
-          typecheckSuccess => testSuccess && lintSuccess && typecheckSuccess
-        )
-      )
+          typecheckSuccess => testSuccess && lintSuccess && typecheckSuccess,
+        ),
+      ),
```
Comment (lostplan) [inline packages/codemirror-graphql/resources/watch.js]: I've had my head out the game for a while. Why add trailing commas?

Comment (acao) [reply packages/codemirror-graphql/resources/watch.js]: it’s on trend now because es2018 i think adoped trailing function param commas

Comment (acao) [reply packages/codemirror-graphql/resources/watch.js]: makes git diffs cleaner, leas merge conflicts. es5 the setting before did it for just literals, all is literally everything else haha

**Comment by acao** [comment]
@lostplan thank you much for reviewing all the commas 😆  glad we got you added! this will save at least 3 unnecessary lines per peer review I promise 🤣 now let's see what new formatting trend will happen next year

**Comment by acao** [comment]
@lostplan do you reccomend squashing or rebasing here? given the one giant prettier commit

**Pull request merged**